hack/code/version_compatibility.go

```diff
@@ -0,0 +1,77 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/aws/karpenter/tools/kompat/pkg/kompat"
+	"github.com/samber/lo"
+)
+
+var uri = "https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html"
```

Comment:
We don't need to pull the version details from a remote endpoint if we're doing manual upgrades, right?